### PR TITLE
Don't allow unserializing classes with a destructor

### PR DESCRIPTION
**Pull request by jderusse**
Prevent destructors with side-effects from being unserialized

### CODE REVIEW

**Comment by keradus** [comment]
were you actually serializing those classes? what is the issue you are eager to solve here?

**Comment by jderusse** [comment]
> were you actually serializing those classes? what is the issue you are eager to solve here?

thanks for reviewing, sorry for the lack of context. We (symfony) fixed yesterday similar cases as security hardening.

I don't serialize these classes (and nobody should). But if these classes are present on the file system, and the application unserialize untrusted data, an attacker could leverage theses classes to execute code via `__destruct`. See https://owasp.org/www-community/vulnerabilities/PHP_Object_Injection.
This PR does not fix the initial issue (unserializing untrusted input) but make the exploit harder for the attacker.

**Comment by keradus** [comment]
thanks for the context!
can you please reference the reasoning in code as well? like John Doe gonna see the code only, and will have no clue why those (and only those) classes forbids serialization

File: src/Cache/FileCacheManager.php [hunk 1]
```diff
@@ -77,6 +77,26 @@ public function __destruct()
         $this->writeCache();
     }
 
+    /**
```
Comment (staabm) [inline src/Cache/FileCacheManager.php:80]: Maybe extract a `ForbiddSerialization` trait, to get the aspect into a single code-unit?

Comment (jderusse) [reply src/Cache/FileCacheManager.php:80]: I'm not sure it worth it...

Comment (keradus) [reply src/Cache/FileCacheManager.php:80]: i'm good with both. I gonna merge as-is and if one is eager to create such trait, please raise a follow-up PR

**Comment by keradus** [comment]
Thank you @jderusse.

**Pull request merged**